esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -422,15 +425,38 @@ def main(cfg):
     """Perform climwip weighting method."""
     models, observations = read_metadata(cfg)
 
-    variable_groups = models.keys()
+    variable_groups_independence = [
+        key for key, value in cfg['independence_contributions'].items()
+        if value > 0
+    ]
+    variable_groups_performance = [
+        key for key, value in cfg['performance_contributions'].items()
+        if value > 0
+    ]
+    assert len(variable_groups_independence) > 0
```

Comment:
Is this necessary? I can imagine you can also calculate weights based on performance only. And if you keep it, would it be nice to write a friendly message for if this assertion fails?

Reply:
Good point updated this to allow either performance only or independence only. (I still exclude both having length one which would lead to equal weighting. In principle we could also allow this but it is a bit of a corner case that is also not useful but makes the code more complex. But I don't have strong opinion on this so let me know if you think this should also be allowed)